cypress/e2e/expenses/Add.cy.js

```diff
@@ -0,0 +1,105 @@
+import '@testing-library/cypress/add-commands';
+
+import {
+  getUKFormattedEuros,
+  getUKFormattedDate,
+  getCurrentMonth,
+} from '../../../src/utils/helpers';
+
+describe('ADD EXPENSE - NOT CURRENT MONTH', () => {
+  before(() => {
+    cy.intercept('GET', '**/items*', []).as('getExpenses');
+    cy.visit('/');
+    cy.wait('@getExpenses');
+  });
+
+  beforeEach(() => {
+    cy.intercept('POST', '**/items*', { fixture: 'expense-post' }).as(
```

Comment:
This is really just to make things a bit more practical and visual. 
We see at the top of the page what our interceptions are but we `cy.wait()` for them specifically in our `it()` blocks.

**In my head, it is like**: _before each_ test block spy on any `POST` request matching the `url`. If they do not happen do nothing. If they happen look if inside the block there is a `cy.wait()`, then respond with the `fixture`.

At the moment we POST only one expense so no problem.
If we had more than one POST or GET requests, with different responses, we would probably need to move the specific POST/GET intercept within the scope of one block but I did not try that yet.

Does it make sense to you guys?

Reply:
My impression is that the POST request will be intercepted in any `it()` block if you stub it inside the `describe()` block. I could be wrong though. If it doesn't work this way, maybe we should consider using the `it()` blocks to describe the app itself ("it adds expenses", "it deletes expenses") instead of the test steps. Then all of our `it()` blocks would be self-contained.

Reply:
I'd leave it like this in this case because, yes, the scope of this test it to `post` only one `expense`. if we will decide to `post` multiple `expenses` with different values, I agree we will need to move the specific `intercept()` into the `it()` block